src/Serilog.HttpClient/LoggingDelegatingHandler.cs

```diff
@@ -203,22 +187,26 @@ static double GetElapsedMilliseconds(long start, long stop)
                     }
                 }
 
-                var responseData = new
-                {
-                    StatusCode = (int?)resp?.StatusCode,
-                    IsSucceed = isRequestOk,
-                    ElapsedMilliseconds = elapsedMs,
-                    BodyString = responseBodyText,
-                    Body = responseBody,
-                    Header = responseHeader,
-                };
-
-                _logger.Write(level, ex, _options.MessageTemplate, new
+                using (Serilog.Context.LogContext.PushProperty("RequestMethod", req.Method))
+                using (Serilog.Context.LogContext.PushProperty("RequestUri", req.RequestUri))
+                using (Serilog.Context.LogContext.PushProperty("RequestScheme", req.RequestUri.Scheme))
+                using (Serilog.Context.LogContext.PushProperty("RequestHost", req.RequestUri.Host))
+                using (Serilog.Context.LogContext.PushProperty("RequestPath", req.RequestUri.AbsolutePath))
+                using (Serilog.Context.LogContext.PushProperty("RequestQueryString", req.RequestUri.Query))
+                using (Serilog.Context.LogContext.PushProperty("RequestQuery", requestQuery))
+                using (Serilog.Context.LogContext.PushProperty("RequestBodyString", requestBodyText))
+                using (Serilog.Context.LogContext.PushProperty("RequestBody", requestBody, destructureObjects: true))
```

Comment:
Also having both RequestBodyString and RequestBody was also a little redundant. RequestBody doesn't deserialize well. It looks like:

```
  "RequestBody": {
    "RootElement": { "ValueKind": "Object", "$type": "JsonElement" },
    "$type": "JsonDocument"
  },
```

I'd opt for only keeping the string version. What do you think?

Reply:
It's because of Destructing Policies that I fixed now. Serializing request, response, query strings as objects is useful when your logs are collected and sent to splunk, elastic or similar products and you can search on body as structured data.
for example in splunk you can search for specific request body like:  "Context.Request.Body.amount"=544700

![image](https://user-images.githubusercontent.com/888192/181214548-6815cdaf-aff9-40e1-afcc-04310aa2d902.png)